### PR TITLE
Update HandlerFactoryImpl to create runtime handlers by schemaVersion

### PR DESCRIPTION
**Pull request by SummerSun**

### CODE REVIEW

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppUtils.java [hunk 1]
```diff
@@ -17,47 +17,49 @@
...
     }
 
     public static void assureWindowsWebApp(final WebApp app) throws MojoExecutionException {
         if (isLinuxWebApp(app)) {
-            throw new MojoExecutionException(JAVA_VERSION_NOT_APPLICABLE);
+            throw new MojoExecutionException(String.format(CONFIGURATION_NOT_APPLICABLE, "Windows"));
+        }
+    }
+
+    public static void assureServerExists(final Server server, final String serverId) throws MojoExecutionException {
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppUtils.java]: The name of `assureServerExists` is confusing. Consider better naming it?

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/configuration/RuntimeSetting.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    public String getOs() {
+        return this.os;
+    }
+
+    public JavaVersion getWindowsJavaVersion() {
+        return StringUtils.isEmpty(javaVersion) ? null : JavaVersion.fromString(javaVersion);
+    }
+
+    public RuntimeStack getLinuxJavaVersion() throws MojoExecutionException {
+        if (StringUtils.isNotEmpty(javaVersion) && StringUtils.equalsIgnoreCase(javaVersion, JRE_8)) {
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/configuration/RuntimeSetting.java]: The `StringUtils.equalsIgnoreCase()` will handle the null check.

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/HandlerFactoryImpl.java [hunk 3]
```diff
@@ -68,13 +75,40 @@ protected RuntimeHandler getV1RuntimeHandler(final AbstractWebAppMojo mojo) thro
...
+                return getV2RuntimeHandlerFromImage(
+                    runtime.getImage(), runtime.getServerId(), runtime.getRegistryUrl());
+            default:
+                throw new MojoExecutionException(
+                    "The value of <os> is unknown, supported values are: windows, linux and docker.");
         }
+    }
 
-        final DockerImageType imageType = WebAppUtils.getDockerImageType(containerSetting);
+    protected RuntimeHandler getV1RuntimeHandlerFromImage(final String imageName, final String serverId,
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/HandlerFactoryImpl.java]: getV1DockerRuntimeHandler()?

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/HandlerFactoryImpl.java [hunk 4]
```diff
@@ -84,14 +118,27 @@ protected RuntimeHandler getV1RuntimeHandler(final AbstractWebAppMojo mojo) thro
...
+                throw new MojoExecutionException(NO_RUNTIME_HANDLER);
         }
-
-        throw new MojoExecutionException(NO_RUNTIME_HANDLER);
     }
 
-    protected RuntimeHandler getV2RuntimeHandler(final AbstractWebAppMojo mojo) throws MojoExecutionException {
-        // todo
-        throw new MojoExecutionException("Unimplemented for schema version: " + mojo.getSchemaVersion());
+    protected RuntimeHandler getV2RuntimeHandlerFromImage(final String image, final String serverId,
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/HandlerFactoryImpl.java]: Same above. The method name is confusing

**Comment by SummerSun** [comment]
#354

**Pull request merged**